### DISCUSSION TITLE
Is port versioning only exposed via manifest?

### DISCUSSION THREAD

**Discussion by jdx-john** [microsoft/vcpkg#20020] [Q&A] [answered]
We have looked at using manifest mode but didn't find it really suited our build flow. However the documentation for versioning (https://github.com/microsoft/vcpkg/blob/master/docs/users/versioning.md, https://github.com/microsoft/vcpkg/blob/master/docs/examples/versioning.getting-started.md) appears to all be couched in terms of manifest files.

Is it possible to control versioning using the 'classic' command-line approach? If so, can anyone point me to documentation or provide an example?

Thankyou.

**Comment by autoantwort** [comment, marked as answer]
No, versioning is only possible in manifest mode :)

**Reply by jdx-john** [reply]
@PhoebeHui so the implication is, all future focus is on manifest mode and everyone who doesn't/can't use vcpkg that way is being abandoned?

**Reply by autoantwort** [reply]
I don't think that classic mode is abandoned, but you can not simply combine it with versioning (if you do, you end up with something like manifest mode. So you need a file that specifies the baseline and all dependencies with versions constraints/overrides). What was your problem with manifest mode? I know you wrote it somewhere, but I couldn't find it :)

**Reply by jdx-john** [reply]
I forget the exact details now but we build using more than one triplet and I think this caused issues. It might have been possible to use it but it became clear we were twisting it in a way it wasn't really designed so we reverted to 'classic' mode.

The issue is if we set up for our next build cycle we need to lock all our ports to whichever version they are today. If we had to set up a new build agent on a clean box for some reason in a month's time - say ours suffers major catastrophe - I know no way to get the same versions we have today.

**Reply by autoantwort** [reply]
If you use msbuild and the vcpkg integration there, it now works because of #19767.

> I know no way to get the same versions we have today.

Port versions are bound to git commits in classic mode. So if you always checkout the same commit in the vcpkg repo you get the same port versions.

**Reply by jdx-john** [reply]
I'm not sure which thing you're referring to re:19767? 
But using git commits as a vcpkg-wide versioning... that is a huge help. I somehow never thought of this but it actually makes locking everyone to a specific set of ports very simple. Thanks for pointing out the obvious :)

**Reply by autoantwort** [reply]
#19767 changed the default installation dir. Before this pr there was one installation dir. So if you have projects with different triplets, it first installed everything for the first triplet, then uninstalls everything and install everything for the next triplet, ... . Now the default installation dir contains the triplet name, so that the different triplet installations don't interact anymore and you can install them side by side. It doesn't installs and uninstalls everything all the time.

**Comment by jdx-john** [comment]
Could this be considered a feature request then?

**Comment by autoantwort** [comment]
I don't think that it will works. Versioning and classic mode cannot be conceptually combined well.
